### ISSUE TITLE
Why webview2 is displayed as a white screen, but other computers can use the same application

### ISSUE THREAD

**Issue by qq386760345** [MicrosoftEdge/WebView2Feedback#3191]
**Description**
On my computer, when webview2 is opened, it is displayed as a white screen, and the webview2 runtime has been installed. But other computers can use the same application

**Version**
SDK: 
Runtime: webview2 runtime version 109.0.1518.78
Framework: Winform
OS: Win10

**Repro Steps**

![image](https://user-images.githubusercontent.com/30264668/216878720-b5967bca-366c-4a4a-b566-733b80878052.png)
![image](https://user-images.githubusercontent.com/30264668/216878741-d2088baf-5f6d-4ce1-96da-bccdeb6fd2c0.png)

**Additional context**

**Comment by sln162** [comment]
![image](https://user-images.githubusercontent.com/11196986/217538676-4cbe7a82-bb71-4a1f-a55b-3631fd788227.png)
初始化的方法中，判断有没有错误，如果没错误，等待130秒，看会不会出现超时错误。--In the initialization method, judge whether there is an error. If there is no error, wait for 130 seconds to see if there will be a timeout error.

**Comment by Huangpuguang** [comment]
I have the same problem, did you solve it?

**Comment by MarkusSchreiner** [comment]
This behaviour also occurs with some of our customers who use our application, since about 3 weeks!
**I would classify this as High Prio**

As far as I know, restarting the application removes this error for some time.
What I also noticed is that the Windows PageFile is relatively large in the moment this behavior occurs (1.9 to 2.5 GB).

We currently shipped our application with WebView2 v1.0.1418.22.
All customers use very latest Edge versions (~110.0.15xx.xx)

**Comment by Shangminx** [comment]
Hi @qq386760345, Have you solved the problem? If not, may you provide some sample code for me to repro this issue? Or you can try to update the runtime version to latest.

**Comment by MarkusSchreiner** [comment]
This problem still occurs with many of our customers.

So far, I have not been able to find a possible cause.
My previous assumption about the PageFile is probably wrong, as the problem also occurs with low 'utilisation' of the PageFile.

**Comment by qq386760345** [comment]
I have discovered a possible reason that when there are certain monitoring software on the device, including but not limited to ESET, it may be loaded together with webview2, which will trigger the integrity check of webview2, resulting in loading exceptions. You can avoid this issue by removing code integrity checks during initialization. The code example is as follows:

```c#
        private async void InitAsync()
        {
            try
            {

                var op = new CoreWebView2EnvironmentOptions("--disable-features=RendererCodeIntegrity");
                var env = await CoreWebView2Environment.CreateAsync(null, "C:\\temp", op);
                await webView21.EnsureCoreWebView2Async(env);
            }
            catch (Exception ex)
            {
                LOGGER.Error("InitAsync webview failed, ex ", ex);
            }
        }```

**Comment by MarkusSchreiner** [comment]
@qq386760345 Thank you - I will give it a try. Good investigation ! 👍

**Comment by sln162** [comment]
@qq386760345   If this is the case as you said, then increase the CoreWebView2_ProcessFailed event, then you should be able to capture the e.ExitCode -1073740760 event, right?

**Comment by MarkusSchreiner** [comment]
@qq386760345 
Hello, I'm sorry to say, that this behaviour still occurs at some of our customers.
And this although I used the workaround with the environment option "--disable-features=RendererCodeIntegrity".

However, I have been able to gather a little more information:

* It seems to be clearly related to one of Edge's render processes.
* When the behaviour occurs, **and I kill one of Edge's running render processes** in the task manager, I receive the error 1 "COREWEBVIEW2_PROCESS_FAILED_KIND_RENDER_PROCESS_EXITED" reported via the ICoreWebView2ProcessFailedEventHandler. 
![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/61575749/c5594a7c-6460-45a1-a781-1fd663e20b99)

* So I can react to this and automatically terminate and restart all existing WebView2 instances in my running application - then the rendering works again!
* What else has been noticed: 
a) Before the white screen appears, the displayed HTML page was still rendered correctly. Only after the user 'interacted' with the page, i.e. marked text with the mouse, did the HTML page suddenly turn white.
b) All subsequent navigation calls in this 'white' WebView2 instance were executed without errors/callbacks - but nothing was displayed.

Unfortunately, I cannot say under which circumstances the bug occurs - it seems to happen very randomly and very rarely.

**Comment by Shangminx** [comment]
Hi @MarkusSchreiner, After you killing the render process and autorestart, did the white screen behaviour recover?

**Comment by MarkusSchreiner** [comment]
> Hi @MarkusSchreiner, After you killing the render process and autorestart, did the white screen behaviour recover?

Yes, it did.

**Comment by beckart** [comment]
I found that an app would have a white box over it on one monitor, but not on the other monitor. I also had a few other people with the same issue, and a lot that were not experiencing any problem. If anyone with the issue changed their monitor scaling to 100% the white box went away.

**Comment by beckart** [comment]
Yes, the issue still exists and I can replicate it with a laptop running Windows 10 and attached monitor.

With the laptop set to scale 150% and a resolution of 1920 X 1200 and the monitor set to scale 100% and resolution 1920 X 1080 the white box will appear on the laptop and moving the app to the monitor results in the white box disappearing. This is not random and I can replicate it using these settings.

The issue has not been significant as

  *   Not a lot of people use the particular process where this is evident.
  *   If a user is using that process they know that a change of resolution will make it usable.
  *   My laptop has recently been updated and using a Windows 11 laptop I can’t recreate the box.

Removing identifying details this is a frame from a CamStudio recording I have. The picture is of the extended display (2 monitors) and the app being on the monitor shows correctly.
***@***.***

Another frame after the move to the laptop monitor. Although difficult to see white on white the box covers the input area.
***@***.***


***@***.*** The form, with the white box over it.

***@***.***The form can be pulled from under the white box, and when that happens the size, and location of the white box changes.

**Comment by criddell** [comment]
We experience this issue as well. For us, we noticed it was happening whenever we were showing a MessageBox while the WebView2 was initializing. If we suppress the message box, it seems to start. Otherwise we get the blank view. If we kill and restart the web view, it starts working.

**Comment by DavidTynewyddBI** [comment]
Any news on this? We recently upgraded our UWP app to use WebView2 and now we have clients experiencing this issue. For some, disabling the integrity check worked initially, but stopped working when they restarted the app. For others it didn't work at all

**Comment by beckart** [comment]
The issue appears to be fixed in Windows 11 however not yet patched in Windows 10. In our case we were using Delphi and as long as I turn “gdiScaling” to “false”, the issue was mitigated as a workaround.

**Comment by jamesjcurrie** [comment]
Some of our users are reporting a similar issue for our Excel addin which uses a WPF Webview2 browser to expose some functionality. The affected users seem to be on Windows 10 and run into the issue intermittently after some time. We are having trouble reproducing it as many of us are on windows 11. Will try some of the reported workarounds to see if there's any improvement.

**Comment by geocine** [comment]
I experience same issue but very similar to this, https://github.com/MicrosoftEdge/WebView2Feedback/issues/4141 . Mine turns white after I leave it open for some time
